paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/sink/partition/PartitionMarkDoneTest.java

```diff
@@ -66,6 +69,107 @@ public void testNotTriggerByCompaction() throws Exception {
         innerTest(false);
     }
 
+    @Test
```

Comment:
I suggest that you also add a IT case like FlinkBatchJobPartitionMarkdoneITCase.

Reply:
+1

Reply:
You were right. I was just kind of tired writing test code. Already write a lot in #5213  -_-!

Reply:
I wiil add ITCase later.

Reply:
Added E2E test in WatermarkPartitionMarkDoneTest#testWaterMarkPartitionMarkDone